### PR TITLE
do not try to upgrade LF < 1.16 contract

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

**Comment by samuel-williams-da** [comment]
I'll review properly on monday, but i noticed you've left a bunch of tests commented out :sweat_smile:

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/UpgradeTest.scala [hunk 1]
```diff
@@ -144,13 +177,17 @@ class UpgradeTest(majorLanguageVersion: LanguageMajorVersion)
   }
 
   val pkgName = {
-    assert(pkg1.name == pkg2.name && pkg2.name == pkg3.name && pkg3.name == pkg4.name)
+    assert(
+      pkg0.metadata.map(
+        _.name
+      ) == pkg1.name && pkg1.name == pkg2.name && pkg2.name == pkg3.name && pkg3.name == pkg4.name
+    )
```
Comment (samuel-williams-da) [inline sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/UpgradeTest.scala:184]: Why not `pkg0.name`? is `pkg0` a different type to `pkg1/2/3/4`?

File: sdk/daml-script/test/daml/upgrades/UpgradesTest.daml [hunk 2]
```diff
@@ -205,12 +210,36 @@ exerciseV1Util p c = p `submit` createAndExerciseExactCmd (V1.ValidUpgradeUtils
...
+tryExerciseV0Util : Choice V0.ValidUpgradeUtils c r => Party -> c -> Script (Either SubmitError r)
+tryExerciseV0Util p c = p `trySubmit` createAndExerciseExactCmd (V0.ValidUpgradeUtils with party = p) c
+
 tryExerciseV1Util : Choice V1.ValidUpgradeUtils c r => Party -> c -> Script (Either SubmitError r)
 tryExerciseV1Util p c = p `trySubmit` createAndExerciseExactCmd (V1.ValidUpgradeUtils with party = p) c
 
 tryExerciseV2Util : Choice V2.ValidUpgradeUtils c r => Party -> c -> Script (Either SubmitError r)
 tryExerciseV2Util p c = p `trySubmit` createAndExerciseExactCmd (V2.ValidUpgradeUtils with party = p) c
 
+mustBeWronglyTypedContract (Left (WronglyTypedContract {})) = pure ()
```
Comment (samuel-williams-da) [inline sdk/daml-script/test/daml/upgrades/UpgradesTest.daml:222]: Please add a type signature

File: sdk/daml-script/test/daml/upgrades/UpgradesTest.daml [hunk 3]
```diff
@@ -205,12 +210,36 @@ exerciseV1Util p c = p `submit` createAndExerciseExactCmd (V1.ValidUpgradeUtils
...
+mustBeWronglyTypedContract  _ = assertFail "Wrong"
+
+fetchNonUpgradableFail : Script ()
+fetchNonUpgradableFail = do
```
Comment (samuel-williams-da) [inline sdk/daml-script/test/daml/upgrades/UpgradesTest.daml:226]: I think mentioning that this is an LF version upgrade/downgrade in the test name would make this a little more clear.
Its hard to see what the test means from the script name

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/UpgradeTest.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/UpgradeTest.scala:184]: `Package#name` is defined the package name iff LF >= 1.16. (as it is use in the scala world only in this case).
`Package#metadata` is always define (except for very old LF <= 8).

So here `pkg0.name = None` while `pkg0.metadata.map(_.name) = Some("-upgrade-test-")`

I admit the name of the former method is maybe not very good,  but do you have another idea ?

Comment (samuel-williams-da) [reply sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/UpgradeTest.scala:184]: Ah fair enough, yeah you'd probably want something like `upgradesPackageName` and have it separate, but for this case its fine.

**Pull request merged**